neo/rawio/intanrawio.py

```diff
@@ -519,19 +679,35 @@ def read_rhd(filename):
     ordered_channels = []
 
     if version >= V("1.2"):
-        data_dtype = [("timestamp", "int32", BLOCK_SIZE)]
+        if file_format == "header-attached":
+            data_dtype = [("timestamp", "int32", BLOCK_SIZE)]
+        else:
+            data_dtype[6] = "int32"
+            channel_number_dict[6] = 1
     else:
-        data_dtype = [("timestamp", "uint32", BLOCK_SIZE)]
+        if file_format == "header-attached":
+            data_dtype = [("timestamp", "uint32", BLOCK_SIZE)]
+        else:
+            data_dtype[6] = "uint32"
+            channel_number_dict[6] = 1
 
     # 0: RHD2000 amplifier channel
     for chan_info in channels_by_type[0]:
         name = chan_info["native_channel_name"]
         chan_info["sampling_rate"] = sr
         chan_info["units"] = "uV"
         chan_info["gain"] = 0.195
-        chan_info["offset"] = -32768 * 0.195
+        if file_format == "header-attached":
+            chan_info["offset"] = -32768 * 0.195
+        else:
+            chan_info["offset"] = 0.0
         ordered_channels.append(chan_info)
-        data_dtype += [(name, "uint16", BLOCK_SIZE)]
+        
+        if file_format == "header-attached":
+            data_dtype += [(name, "uint16", BLOCK_SIZE)]
+        else:
+            data_dtype[0] = "int16"
```

Comment:
@ChauhanT is this what you mean?

Reply:
Sorry, my bad. Two things I had not checked : 
1. The interface proposed by neo. I thought every reader was doing its own thing.
2. I hadn't realised it's an offset as @zm711 has pointed out below.

Reply:
Ah, OK. Yes, if you look at the bottom of the headers that you printed above you can see that they have dtypes. Those dtypes indicate what every column means. 

Also, taking this asside. in a spikeinterface object you can get the header like this:

```
recording_extractor.neo_reader.header
```

All the best,

Reply:
Ah, this code that I am referencing here addressess your last question btw:

https://github.com/NeuralEnsemble/python-neo/pull/1402#issuecomment-1979427022

Header-attached is what you call "one-file-per-recording" (which is also a good name btw!). So here, we have a different dtype depending on the format.

Reply:
Many thanks ! Really appreciate the help from both.

Reply:
Thanks again @h-mayorquin . I think the datatype is definitely read in correctly, because spikeinterface has the correct dtype. I'll try to work out why the header still reports it as uint.